src/thunderbots/software/ai/hl/stp/stp.cpp

```diff
@@ -0,0 +1,102 @@
+#include "ai/hl/stp/stp.h"
+
+#include <ai/hl/stp/play/play_factory.h>
+#include <munkres/munkres.h>
+
+#include <chrono>
+#include <exception>
+#include <random>
+
+#include "ai/hl/stp/play/play.h"
+#include "ai/hl/stp/tactic/tactic.h"
+#include "ai/intent/move_intent.h"
+#include "util/logger/init.h"
+
+STP::STP(long random_seed)
+    : random_number_generator(random_seed), uniform_distribution(1, 6)
+{
+}
+
+std::vector<std::unique_ptr<Intent>> STP::getIntentAssignment(const World& world)
+{
+    // Assign a new play if we don't currently have a play assigned, the current play's
+    // invariant no longer holds, or the current play is done
+    if (!current_play || !current_play->invariantHolds(world) || current_play->done())
+    {
+        try
+        {
+            current_play = std::move(calculateNewPlay(world));
+        }
+        catch (const std::runtime_error& e)
+        {
+            LOG(WARNING) << "Unable to assign a new Play. No Plays are valid"
+                         << std::endl;
+            // TODO: Set current_play to a reasonable default, like our Stop play
+            // https://github.com/UBC-Thunderbots/Software/issues/410
+        }
+    }
+
+    // Run the current play
+    auto tactics = current_play->getTactics(world);
+
+    std::vector<std::unique_ptr<Intent>> intents;
+    if (tactics)
+    {
+        // Assign robots to the tactics
+        auto assigned_tactics = assignRobotsToTactics(world, *tactics);
+        // Get the Intent each tactic wants to run
+        for (const auto& tactic : assigned_tactics)
+        {
+            auto intent = tactic->getNextIntent();
+            if (intent)
+            {
+                intents.emplace_back(std::move(intent));
+            }
+        }
+    }
+
+    return intents;
+}
+
+std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
+    const World& world, std::vector<std::shared_ptr<Tactic>> tactics) const
+{
+    return {};
```

Comment:
We probably want to implement this?

Reply:
Being done in a separate PR to keep the PR diffs managebale